var/spack/repos/builtin/packages/graphviz/package.py

```diff
@@ -99,26 +101,40 @@ def configure_args(self):
         # These language bindings have not yet been tested.  They
         # likely need additional dependencies to get working.
         untested_bindings = (
-            '+swig', '+sharp', '+go', '+guile', '+io',
+            '+sharp', '+go', '+guile', '+io',
             '+lua', '+ocaml', '+php',
             '+python', '+r', '+ruby', '+tcl')
 
         for var in untested_bindings:
             if var in spec:
-                raise SpackException(
+                raise InstallError(
                     "The variant {0} for language bindings has not been "
                     "tested.  It might or might not work.  To try it "
                     "out, run `spack edit graphviz`, and then move '{0}' "
                     "from the `untested_bindings` list to the "
                     "`tested_bindings` list.  Be prepared to add "
                     "required dependencies.  "
                     "Please then submit a pull request to "
-                    "http://github.com/llnl/spack")
+                    "http://github.com/llnl/spack".format(var))
+
+        need_swig = True if ("+swig" in spec) else False
 
         for var in tested_bindings:
-            enable = 'enable' if (var in spec) else 'disable'
+            if (var in spec):
+                need_swig = True
```

Comment:
since you override `need_swig`, is not it better to move this loop outside and use `conflicts` statement to say that each `+var` among `tested_bindings` conflicts with `~swig`?